plugin/updater_gpu/src/device_helpers.cuh

```diff
@@ -590,7 +590,7 @@ __global__ void launch_n_kernel(int device_idx, size_t begin, size_t end,
 template <int ITEMS_PER_THREAD = 8, int BLOCK_THREADS = 256, typename L>
 inline void launch_n(int device_idx, size_t n, L lambda) {
   safe_cuda(cudaSetDevice(device_idx));
-  const int GRID_SIZE = div_round_up(n, ITEMS_PER_THREAD * BLOCK_THREADS);
+  const int GRID_SIZE = div_round_up(n, ITEMS_PER_THREAD * BLOCK_THREADS); // FIXME: If n is large enough, this limits size can do.  Need to use 2D or 3D grid?
```

Comment:
Ok, I added a note there.  Not a bad limit currently for real data, but can be fixed in future.